### ISSUE TITLE
Predicted Depth to Colorful Point Cloud

### ISSUE THREAD

**Issue by YuxuanSnow** [prs-eth/Marigold#6]
Hi,

nice work! I always want to compare the predicted depth with the colorful unprojected point cloud. I compared Marigold, ZoeDepth, OmniDatav2. I tried the following image.

![example_5](https://github.com/prs-eth/Marigold/assets/50771152/e8f50bf2-70b1-4d5b-a4f1-81759c4f4e4b)

Marigold:
![image](https://github.com/prs-eth/Marigold/assets/50771152/31e56261-d2e4-47bb-b274-c13e32d17f67)

OmniDatav2:
![image](https://github.com/prs-eth/Marigold/assets/50771152/6ed911f7-b860-4c5f-8d9f-492868e7eecf)

ZoeN:
![image](https://github.com/prs-eth/Marigold/assets/50771152/67bfd095-b490-4415-8e5e-1bcc6836358e)

It's interesting that previous method predicts more likely to a flat geometry, while Marigold can preserve it better :-)

**Comment by YuxuanSnow** [comment]
However, i think my unprojection function is not valid for some scenario such as indoor room. 
![4780_rgb](https://github.com/prs-eth/Marigold/assets/50771152/9883c4de-7994-4f5e-929c-1bb99c2b8f30)

As you can see, the point cloud doesn't have a flat floor and ceiling.
![image](https://github.com/prs-eth/Marigold/assets/50771152/a66e8c07-9a6f-48a9-a0af-9d58e4dcc3ff)

I believe the reason is because of the normalization scale of the predicted depth range. Can you also provide the unprojection function that you used to generate point cloud in the paper?

**Comment by sanskar107** [comment]
I am also facing the same issue, the pointcloud looks distorted on indoor images.

**Comment by coccofresco** [comment]
@YuxuanSnow what expression did you use? is not perfect, but not so bad (far better than mine, for sure)

**Comment by nandometzger** [comment]
Hi,

Thank you for your interest in Marigold and the generation of 3D point clouds from it. 

Please note that Marigold predicts affine invariant depth, not metric depth. Meaning that even if the the camera intrinsics are known. The unprojection function still is undefined by a scale and a shift. For the example of @YuxuanSnow the scale and shift are not chosen appropriately. 

We quickly described our workflow in the Appendix Section 4. In practice, it means that you have to try interatively try configurations of scale and shifts, and pick the visually best one.

Hope that helps. Feel free to ask if you have questions.

**Comment by coccofresco** [comment]
> Hi,
> 
> Thank you for your interest in Marigold and the generation of 3D point clouds from it.
> 
> Please note that Marigold predicts affine invariant depth, not metric depth. Meaning that even if the the camera intrinsics are known. The unprojection function still is undefined by a scale and a shift. For the example of @YuxuanSnow the scale and shift are not chosen appropriately.
> 
> We quickly described our workflow in the Appendix Section 4. In practice, it means that you have to try interatively try configurations of scale and shifts, and pick the visually best one.
> 
> Hope that helps. Feel free to ask if you have questions.

Hi

Thank you very much for your reply.
Unfortunately, I am more of a 3d modeler than a programmer or researcher, so even going by trial and error is extremely complicated for me.
Would it be possible to have a basic equation where it would be easy to intervene in the scaling and shift factors?

I understand that it transcends the purpose of the research, but it would then be great in the future to have a built-in tool with which to play with parameters while viewing the point cloud, and then export a linearized depth map. 

Thanks in advance

**Comment by coccofresco** [comment]
thank you very much, i was able to get it working in my 3d editor. The amount of information you can retrieve is impressive. Even on a frontal human face, it can retrieve a profile that is not perfect, but functional for quick retouching. Really amazing.

**Comment by toshas** [comment]
@coccofresco Thank you! If the data and time permit, could you please showcase some of the results on Twitter and refer our original announcement post? https://twitter.com/AntonObukhov1/status/1732946419663667464?t=8iIVVDbbrhwGHQgHNOoj2A&s=19

**Comment by Haozong-Zeng** [comment]
> affine invariant depth

Thanks for the great work. 

I was wondering if I have some prior knowledges on the scale and location (both object & camera) of the object I want predict (e.g. a rough point cloud, estimated 3D bounding-box, etc.), will it be possible to adjust Marigold prediction? 

Also I wonder if the training script will be released in the future for fine-tuning the model.

**Comment by fradif96** [comment]
@YuxuanSnow Could you please share the snippet code that you used for point cloud generation and visualization? 
Thanks in advance

**Comment by PHOENIXFURY007** [comment]
> Hi,
> 
> nice work! I always want to compare the predicted depth with the colorful unprojected point cloud. I compared Marigold, ZoeDepth, OmniDatav2. I tried the following image.
> 
> ![example_5](https://private-user-images.githubusercontent.com/50771152/289315599-e8f50bf2-70b1-4d5b-a4f1-81759c4f4e4b.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNzU5NzEsIm5iZiI6MTcxMjE3NTY3MSwicGF0aCI6Ii81MDc3MTE1Mi8yODkzMTU1OTktZThmNTBiZjItNzBiMS00ZDViLWE0ZjEtODE3NTljNGY0ZTRiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDIwMjExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiYTVkNjkzYjdlNmUzNDdjNTQ1N2ZmOGRkN2MzY2NkYmZkNGY3MWZjZGIyYThhNGU0NmFhNTU0NDk5MmMxZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h3CHD7m5SJUvVyKfBfEj0cu5tT_Y3TrGmgRg1bg_7B4)
> 
> Marigold: ![image](https://private-user-images.githubusercontent.com/50771152/289316216-31e56261-d2e4-47bb-b274-c13e32d17f67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNzU5NzEsIm5iZiI6MTcxMjE3NTY3MSwicGF0aCI6Ii81MDc3MTE1Mi8yODkzMTYyMTYtMzFlNTYyNjEtZDJlNC00N2JiLWIyNzQtYzEzZTMyZDE3ZjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDIwMjExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MWJiYjQ4ZWMxY2VjMzlkN2NiYzM5YTQ4MjBkMDQ4OTc2OWJiZTE5ZWNhOTA1ZjgzZjY0OTgwZTQ0NTNlYTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.a4V6JkAcw0WYUEW3XVD7ZdegnonwsMYJFFtrGwSz-OU)
> 
> OmniDatav2: ![image](https://private-user-images.githubusercontent.com/50771152/289316177-6ed911f7-b860-4c5f-8d9f-492868e7eecf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNzU5NzEsIm5iZiI6MTcxMjE3NTY3MSwicGF0aCI6Ii81MDc3MTE1Mi8yODkzMTYxNzctNmVkOTExZjctYjg2MC00YzVmLThkOWYtNDkyODY4ZTdlZWNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDIwMjExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0MThkMTFkNjM3Mzg1ZDgzMzYwNDhmYjUyZjM0MzhlMWQzMGZiYjFhZmEwYTIwZmQ1MWVlYmUwMGI2ZDkxODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LN-LuPB0lCMZbRcHxITd68s2HrmocXjvEnKiLHvUD_o)
> 
> ZoeN: ![image](https://private-user-images.githubusercontent.com/50771152/289316192-67bfd095-b490-4415-8e5e-1bcc6836358e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNzU5NzEsIm5iZiI6MTcxMjE3NTY3MSwicGF0aCI6Ii81MDc3MTE1Mi8yODkzMTYxOTItNjdiZmQwOTUtYjQ5MC00NDE1LThlNWUtMWJjYzY4MzYzNThlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDAzVDIwMjExMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1ZGM3YzY2N2QzMDljYmYzZjliMGM5M2NmMWY4N2U3MTE1ZGNiMzk0MzJlOGUwOGJjNDVjMTRiMjM2Y2Q2MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ynL51sqBtKChb5TBlBAEgmXeex-38nTapIPWJ_mHaZ0)
> 
> It's interesting that previous method predicts more likely to a flat geometry, while Marigold can preserve it better :-)

Hello, 
Can you share how to generate the point clouds from the depth maps? And also what tool are you using to visualize the  3d point cloud?
